wiki/NSHM-2022.md

```diff
@@ -0,0 +1,236 @@
+╭─ Options ─────────────────────────────────────────────────────────────────────────────────────────────╮
+│ --excd-rates        FLOAT  List of exceedance rates to compute UHS for. One of `excd_rates` or `rps`  │
+│                            must be provided.                                                          │
+│                            [default: None]                                                            │
+│ --rps               FLOAT  List of return periods to compute UHS for. One of `excd_rates` or `rps`    │
+│                            must be provided.                                                          │
+│                            [default: None]                                                            │
+│ --help                     Show this message and exit.                                                │
+╰───────────────────────────────────────────────────────────────────────────────────────────────────────╯
+```
+
+## Disaggregation
+
+The disaggregation is run directly via OQ, and takes a significant amount of time (~6 hours on `hypocentre`) and memory. 
+Memory usage is generally the main limiting factor, and can easily exceed 100Gb, depending on the disagg bin specification. 
+Would not recommend trying to run on a standard desktop, `hypocentre` is generally a good choice.
+
+### Setup
+- Create a run directoy
+- Add the NSHM GMM logic tree file,  `NZ_NSHM_GMM_LT_final_EE.xml`
+- Add the source model file, `source_model.xml`
+- Add the `source_definitions` folder (generally via a symbolic link)
+- Add the `site.csv` 
+- Add the `disagg_job.ini`
+
+See [here](https://www.dropbox.com/scl/fo/u4l15xrz10vsmbg6jqhtk/ABSTE60eiLjyTg9Wey0peDU?rlkey=56xvazcl8lys1wx4hfg5bt4tj&st=ssue8t12&dl=0) for the GMM logic tree, source model file and source definitions folder.
+
+**Example `site.csv`**
+```
+lon,lat,vs30,z1pt0,z2pt5,backarc
+172.63,-43.53,400.0,355.7170357122867,1.2646109912757877,False
+```
+
+**Disagg Configuration**
+All configuration is done in the `disagg_job.ini` file, an example is shown below:
+```ini
+[general]
+description = Disaggregation
+random_seed = 25
+calculation_mode = disaggregation
+ps_grid_spacing = 30
+
+[logic_tree]
+number_of_logic_tree_samples = 0
+
+[erf]
+rupture_mesh_spacing = 4
+width_of_mfd_bin = 0.1
+complex_fault_mesh_spacing = 10.0
+area_source_discretization = 10.0
+
+[site_params]
+site_model_file = ./sites.csv
+
+[calculation]
+source_model_logic_tree_file = ./source_model.xml
+gsim_logic_tree_file = ./NZ_NSHM_GMM_LT_final_EE.xml
+investigation_time = 1.0
+truncation_level = 4
+maximum_distance = {'Active Shallow Crust': [(4.0, 0), (5.0, 100.0), (6.0, 200.0), (9.5, 300.0)],
+        'Subduction Interface': [(5.0, 0), (6.0, 200.0), (10, 500.0)],
+        'Subduction Intraslab': [(5.0, 0), (6.0, 200.0), (10, 500.0)]}
+
+[disagg]
+max_sites_disagg = 1
+mag_bin_width = 0.199999
```

Comment:
why can't this be 0.2?

Reply:
It can be, but it causes issues down the line with plotting from what I remember.